app/controllers/games_controller.rb

```diff
@@ -27,6 +27,7 @@ def update
     if @game.valid?
       current_user.games << game
       redirect_to @game
+      @game.populate_board if @game.pieces.empty?
```

Comment:
Not sure if this is the cause, but you should move this line to right above `redirect_to @game`. You want to populate the board before sending the user to the page that displays the board.